include/basefunctions.hxx

```diff
@@ -72,6 +73,51 @@ inline ROOT::RDF::RNode rename(ROOT::RDF::RNode df,
                                const std::string &outputname) {
     return df.Define(outputname, [](const T &q) { return q; }, {inputname});
 }
+
+/// Function to writeout a variable from a particle. The particle
+/// is identified via the index stored in the input vector
+///
+/// \param df the dataframe to add the quantity to
+/// \param outputname name of the new column containing the variable value
+/// \param position index of the position in the input vector
+/// \param vecname name of the column containing the input vector
+/// \param column name of the column containing the variable values
+///
+/// \returns a dataframe with the new column
+
+template <typename T>
+ROOT::RDF::RNode getvar(ROOT::RDF::RNode df, const std::string &outputname,
+                        const int &position, const std::string &vecname,
+                        const std::string &column) {
+    return df.Define(
+        outputname,
+        [position](const ROOT::RVec<int> &vec, const ROOT::RVec<T> &col) {
+            const int index = vec.at(position);
```

Comment:
I think it would make sense to return an error if the `position` is invalid. Here the `vec` is a vector of indices of selected leptons that play the same role as `dileptonpair` for dilepton channels.

Reply:
yeah that is true, you will never actually readout a quantity without applying any kind of selection to it, so I agree to keep this in "crashing" mode. Maybe still add an error message like here: https://github.com/KIT-CMS/CROWN/blob/c76b94eb07ca8ce4e740874d52a0fa10b0e5140b/src/lorentzvectors.cxx#L58 (longterm this should also be done for other cases in the framework as well)